internal/loader/configuration_setting_loader.go

```diff
@@ -386,7 +387,10 @@ func (csl *ConfigurationSettingLoader) ResolveSecretReferences(
 						}
 						lock.Lock()
 						defer lock.Unlock()
-						resolvedSecretReferences[name].Data[currentKey] = []byte(*resolvedSecret.Value)
+						resolvedSecrets[name].Data[currentKey] = []byte(*resolvedSecret.Value)
+						currentUriSegment := targetSecretReference.SecretsMetadata[currentKey]
```

Comment:
Maybe I just didn't get what you are trying to do here, why don't just do:

```
targetSecretReference.SecretsMetadata[currentKey].SecretId= resolvedSecret.ID

```

Reply:
In Golang, cannot assign to struct field in `map[string]*targetSecretReference`

Reply:
I see